### ISSUE TITLE
Incorrect work of look_at function

### ISSUE THREAD

**Issue by Terinyo** [godotengine/godot#89578]
### Tested versions

- Reproducible in: 4.2-dev4 and later including 4.3-dev5
- Not reproducible in: 4.2-dev3 and earlier 

### System information

Windows 10 - Godot v4.2.1.stable - Compatibility

### Issue description

The bug occurs when:
- "Compatibility" rendering method is used
- "display/window/stretch/scale_mode" is an integer
- "display/window/stretch/mode" is viewport or canvas_item.
- "display/window/size/viewport" size does not equal the window size.

The bug manifests as if the rotation angle is being calculated not from the position of the node, but from some point with an offset from the node. This offset correlates with the difference between the viewport size in the project settings and the window size.

### Steps to reproduce

To reproduces the bug, you can use the `look_at(get_global_mouse_position())` function in any node, such as Sprite2D.

### Minimal reproduction project (MRP)

[Bug.zip](https://github.com/godotengine/godot/files/14624346/Bug.zip)

**Comment by kleonc** [comment]
The issue happens for `display/window/stretch/scale_mode == integer` when "desired" viewport height is greater than the window size. In such case the viewport size is forced to the desired size (integer scale of `1`) so it ends up bigger than the window. However, such viewport ends up being render-wise attached to the window's bottom-left corner, while it seems all other calculations assume it's attached to the window's top-left corner. Hence the discrepancy equal to the height difference between the window/viewport sizes.

![zguDuR3qgv](https://github.com/godotengine/godot/assets/9283098/f0ae2a2f-15d0-4e94-a126-a7d444d64acf)|![N4VeNOQNFF](https://github.com/godotengine/godot/assets/9283098/7a9a5b0f-535d-4304-a202-8f4ae091a1e8)|![3mv50jPLeQ](https://github.com/godotengine/godot/assets/9283098/df3a073e-72e8-4d28-b342-95a92765634f)|![Qf8mR3h4oC](https://github.com/godotengine/godot/assets/9283098/e1df4d09-6061-49ef-8f96-4def76ef01c1)
-|-|-|-

MRP changed to produce the above: [Bug2.zip](https://github.com/godotengine/godot/files/14625241/Bug2.zip)


Happens since the addition in #75784 (cc @Riteo).

**Comment by Sauermann** [comment]
To me, it is not clear, what the bug is exactly:
1. The Viewport should not be attached to the bottom-left corner, but to the top-left; the input-coordinates are correct
2. The Viewport position at the bottom-left corner is correct; input-coordinates need to be adjusted

**Comment by Sauermann** [comment]
I agree, that the Viewport should be attached to the top-left corner.
I believe, that `Window::_update_viewport_size` would be a good place to start investigating this. Not sure, but perhaps then this leads to `RenderingServer::viewport_attach_to_screen`.
